<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MaemoDeviceConfigurationsSettingsWidget</class>
 <widget class="QWidget" name="MaemoDeviceConfigurationsSettingsWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>602</width>
    <height>421</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Maemo Device Configurations</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QFormLayout" name="formLayout">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="configurationLabel">
         <property name="text">
          <string>&amp;Configuration:</string>
         </property>
         <property name="buddy">
          <cstring>configurationComboBox</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="configurationComboBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="editable">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QFrame" name="frame">
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QWidget" name="detailsWidget" native="true">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <layout class="QFormLayout" name="formLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="deviceNameLabel">
             <property name="text">
              <string>&amp;Name</string>
             </property>
             <property name="buddy">
              <cstring>nameLineEdit</cstring>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="nameLineEdit"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="deviceTypeLabel">
             <property name="text">
              <string>Device type:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QWidget" name="deviceTypeButtonsWidget" native="true">
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <property name="margin">
               <number>0</number>
              </property>
              <item>
               <widget class="QRadioButton" name="deviceButton">
                <property name="text">
                 <string>Remote device</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="simulatorButton">
                <property name="text">
                 <string>Maemo &amp;emulator</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="authTypeLabel">
             <property name="text">
              <string>Authentication type:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QWidget" name="authTypeButtonsWidget" native="true">
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <property name="margin">
               <number>0</number>
              </property>
              <item>
               <widget class="QRadioButton" name="passwordButton">
                <property name="text">
                 <string>Password</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="keyButton">
                <property name="text">
                 <string>&amp;Key</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="hostNameLabel">
             <property name="text">
              <string>&amp;Host name:</string>
             </property>
             <property name="buddy">
              <cstring>hostLineEdit</cstring>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLineEdit" name="hostLineEdit">
             <property name="placeholderText">
              <string>IP or host name of the device</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="sshPortLabel">
             <property name="text">
              <string>&amp;SSH port:</string>
             </property>
             <property name="buddy">
              <cstring>sshPortSpinBox</cstring>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QSpinBox" name="sshPortSpinBox">
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="maximum">
                <number>65535</number>
               </property>
               <property name="value">
                <number>22</number>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="freePortsLabel">
             <property name="text">
              <string>Free ports:</string>
             </property>
             <property name="buddy">
              <cstring>portsLineEdit</cstring>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QLineEdit" name="portsLineEdit">
               <property name="toolTip">
                <string>You can enter lists and ranges like this: 1024,1026-1028,1030</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="portsWarningLabel">
               <property name="text">
                <string>TextLabel</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="connectionTimeoutLabel">
             <property name="text">
              <string>Connection time&amp;out:</string>
             </property>
             <property name="buddy">
              <cstring>timeoutSpinBox</cstring>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QSpinBox" name="timeoutSpinBox">
               <property name="suffix">
                <string>s</string>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
               <property name="singleStep">
                <number>10</number>
               </property>
               <property name="value">
                <number>1000</number>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="userNameLabel">
             <property name="text">
              <string>&amp;Username:</string>
             </property>
             <property name="buddy">
              <cstring>userLineEdit</cstring>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QLineEdit" name="userLineEdit"/>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="passwordLabel">
             <property name="text">
              <string>&amp;Password:</string>
             </property>
             <property name="buddy">
              <cstring>pwdLineEdit</cstring>
             </property>
            </widget>
           </item>
           <item row="8" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QLineEdit" name="pwdLineEdit">
               <property name="echoMode">
                <enum>QLineEdit::Password</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="showPasswordCheckBox">
               <property name="text">
                <string>Show password</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="9" column="0">
            <widget class="QLabel" name="keyLabel">
             <property name="text">
              <string>Private key file:</string>
             </property>
            </widget>
           </item>
           <item row="9" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="Utils::PathChooser" name="keyFileLineEdit" native="true"/>
             </item>
             <item>
              <widget class="QPushButton" name="makeKeyFileDefaultButton">
               <property name="text">
                <string>Set as Default</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QPushButton" name="addConfigButton">
       <property name="text">
        <string>&amp;Add</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="removeConfigButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>&amp;Remove</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="testConfigButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>Click here to check whether this device is properly set up to run Maemo projects.</string>
       </property>
       <property name="text">
        <string>&amp;Test</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="generateKeyButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>Click here if you don't have an SSH key yet.</string>
       </property>
       <property name="text">
        <string>&amp;Generate SSH Key ...</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="deployKeyButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>This will enable you to log into the device without a password.</string>
       </property>
       <property name="text">
        <string>&amp;Deploy Public Key ...</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="remoteProcessesButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>Click here to see which processes are running on the device.</string>
       </property>
       <property name="text">
        <string>Remote Processes ...</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Utils::PathChooser</class>
   <extends>QWidget</extends>
   <header location="global">utils/pathchooser.h</header>
   <container>1</container>
   <slots>
    <signal>editingFinished()</signal>
    <signal>browsingFinished()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>deviceButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>deviceTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>231</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>511</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hostLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>hostNameEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>335</x>
     <y>148</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>userLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>userNameEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>430</x>
     <y>274</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pwdLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>passwordEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>306</x>
     <y>302</y>
    </hint>
    <hint type="destinationlabel">
     <x>423</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>simulatorButton</sender>
   <signal>clicked(bool)</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>deviceTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>335</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>87</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addConfigButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>addConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>465</x>
     <y>27</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>118</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeConfigButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>deleteConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>473</x>
     <y>61</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>101</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>passwordButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>authenticationTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>240</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>keyFileLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>keyFileEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>keyFileLineEdit</sender>
   <signal>browsingFinished()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>keyFileEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>257</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>testConfigButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>testConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>473</x>
     <y>90</y>
    </hint>
    <hint type="destinationlabel">
     <x>428</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deployKeyButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>deployKey()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>473</x>
     <y>148</y>
    </hint>
    <hint type="destinationlabel">
     <x>510</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>keyButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>authenticationTypeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>280</x>
     <y>135</y>
    </hint>
    <hint type="destinationlabel">
     <x>525</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nameLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>configNameEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>237</x>
     <y>75</y>
    </hint>
    <hint type="destinationlabel">
     <x>330</x>
     <y>332</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>generateKeyButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>showGenerateSshKeyDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>585</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>358</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>timeoutSpinBox</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>timeoutEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>217</x>
     <y>245</y>
    </hint>
    <hint type="destinationlabel">
     <x>6</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>timeoutSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>timeoutEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>217</x>
     <y>245</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sshPortSpinBox</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>sshPortEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>210</x>
     <y>189</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>211</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sshPortSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>sshPortEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>210</x>
     <y>189</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>136</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showPasswordCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>showPassword(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>429</x>
     <y>301</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>262</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>portsLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>handleFreePortsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>259</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>393</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>remoteProcessesButton</sender>
   <signal>clicked()</signal>
   <receiver>MaemoDeviceConfigurationsSettingsWidget</receiver>
   <slot>showRemoteProcesses()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>591</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>598</x>
     <y>378</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>configNameEditingFinished()</slot>
  <slot>deviceTypeChanged()</slot>
  <slot>hostNameEditingFinished()</slot>
  <slot>sshPortEditingFinished()</slot>
  <slot>timeoutEditingFinished()</slot>
  <slot>userNameEditingFinished()</slot>
  <slot>passwordEditingFinished()</slot>
  <slot>addConfig()</slot>
  <slot>selectionChanged()</slot>
  <slot>deleteConfig()</slot>
  <slot>authenticationTypeChanged()</slot>
  <slot>keyFileEditingFinished()</slot>
  <slot>testConfig()</slot>
  <slot>deployKey()</slot>
  <slot>gdbServerPortEditingFinished()</slot>
  <slot>currentConfigChanged(int)</slot>
  <slot>showGenerateSshKeyDialog()</slot>
  <slot>showPassword(bool)</slot>
  <slot>handleFreePortsChanged()</slot>
  <slot>showRemoteProcesses()</slot>
 </slots>
</ui>
